File: RefactoringExercise/RefactoringExerciseNode/service/countryRepository.js
"use strict";

const countryRepository = function () {
    this.getCountries = function () {
        let countries = [
            {
                countryId: 1,
                name: "United States",
                shortName: "US",
                postalCodeRequired: true,
                stateProvinces: [
                    {
                        name: "Colorado",
                        shortName: "CO",
                        stateProvinceId: 6
                    },
                    {
                        name: "Nevada",
                        shortName: "NV",
                        stateProvinceId: 28
                    }
                ]
            },
            {
                countryId: 2,
                name: "Canada",
                shortName: "CA",
                postalCodeRequired: false,
                stateProvinces: [
                    {
                        name: "Quebec",
                        shortName: "Quebec",
                        stateProvinceId: 52
                    }
                ]
            }
        ];
        return countries;
    };
};
exports.CountryRepository = countryRepository;

File: RefactoringExercise/RefactoringExercise/CountryRepository.cs
using System.Collections.Generic;

namespace RefactoringExercise
{
    public class CountryRepository
    {
        public List<Country> GetCountries()
        {
            var countries = new List<Country>
            {
                new Country
                {
                    CountryId = 1,
                    Name = "United States",
                    ShortName = "US",
                    StateProvinces = new List<StateProvince>
                    {
                        new StateProvince
                        {
                            Name = "Colorado",
                            ShortName = "CO",
                            StateProvinceId = 6
                        },
                        new StateProvince
                        {
                            Name = "Nevada",
                            ShortName = "NV",
                            StateProvinceId = 28
                        }
                    }
                },
                                new Country
                {
                    CountryId = 2,
                    Name = "Canada",
                    ShortName = "CA",
                    StateProvinces = new List<StateProvince>
                    {
                        new StateProvince
                        {
                            Name = "Quebec",
                            ShortName = "Quebec",
                            StateProvinceId = 52
                        }
                    }
                }
            };

            return countries;
        }
    }
}
